lib/srv/discovery/status.go

```diff
@@ -293,5 +300,232 @@ func (s *Server) ReportEC2SSMInstallationResult(ctx context.Context, result *ser
 
 	s.updateDiscoveryConfigStatus(result.DiscoveryConfig)
 
+	s.awsEC2Tasks.addFailedEnrollment(
```

Comment:
What's the difference between this invocation and the one in discovery.go above? Just trying to understand what errors will be captured by this function vs the other one cause they seem to be both processing results of SSM runs?

Reply:
We have 3 places where we generate tasks:
A) we describe the Instance's SSM Agent state and if there's an error (eg, ssm not installed, ssm not online, os not supported)
B) sending the command (`ssm:SendCommand`) fails (wrong permissions, document params issues, ...)
C) after starting the execution, we check the execution result for each instance

A and C are reported using the `ReportEC2SSMInstallationResult`
B is reported in `discovery.go`
https://github.com/gravitational/teleport/blob/aeefe42e8885355843a7e5803e439a3036abb8d2/lib/srv/discovery/discovery.go#L963